src/js/components/MapPanelLocationBar.jsx

```diff
@@ -73,14 +98,18 @@ export default class MapPanelLocationBar extends React.Component {
                 // But we also have the bonus of not needing to make a reverse geocode request
                 // for small changes of the map center.
                 if (delta > MAP_UPDATE_DELTA) {
-                    this.reverseGeocode(currentLatLng);
-                    this.setState({
-                        bookmarkActive: false,
-                        latlng: {
-                            lat: currentLatLng.lat,
-                            lng: currentLatLng.lng
-                        }
-                    });
+                    this.reverseGeocode(currentLatLng)
+                        .then((state = {}) => {
```

Comment:
is this because you save on having to call the map here?
```
latlng: {
    lat: mapCenter.lat.toFixed(latlngLabelPrecision),
    lng: mapCenter.lng.toFixed(latlngLabelPrecision)
    lat: 0,
    lng: 0
} // Represents lat lng of current position of the map
```